pandas/tests/groupby/test_apply.py

```diff
@@ -657,3 +657,34 @@ def test_apply_with_mixed_types():
 
     result = g.apply(lambda x: x / x.sum())
     tm.assert_frame_equal(result, expected)
+
+
+def test_apply_datetime_issue():
+    # GH-28247
+    # groupby-apply throws an error if one of the columns in the DataFrame
+    #   is a datetime object and the column labels are different from
+    #   standard int values in range(len(num_columns))
+
+    df = pd.DataFrame({"a": ["foo"], "b": [datetime.today()]})
```

Comment:
can you parameterize these tests, only thing changing is the b column which you can pass in the paramters